### PR TITLE
add version manifests and schemas

### PR DESCRIPTION
**Pull request by omissis**

### CODE REVIEW

File: furyctl-defaults.yaml [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+        # if not set, no resource patch will be created
+        resources:
+          requests:
+            cpu: ""
+            memory: ""
+          limits:
+            cpu: ""
+            memory: ""
+        # if set, it will override the volumeClaimTemplates in the opensearch statefulSet
+        storage_request: ""
```
Comment (lnovara) [inline furyctl-defaults.yaml:91]:
```suggestion
        storageSize: 150Gi
```

File: furyctl-defaults.yaml [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+          # can be certManager, secret or none
+          provider: certManager # it uses the configuration below as default when certManager is chosen
+          secret: #if we want to use custom certificates, the template should create a secret and set it as default certificate in NGINX, so patch nginx deployment accordingly
+            cert: |
+              value
+            key: |
+              value
+            ca: |
+              value
+      # the standard configuration for cert-manager on the ingress module
```
Comment (lnovara) [inline furyctl-defaults.yaml:53]: Both fields are immutable.
```suggestion
      externalDns:
        privateIamRoleArn: ""
        publicIamRoleArn: ""
      # the standard configuration for cert-manager on the ingress module
```

**Review by lnovara — COMMENTED** [review]
Together with @ralgozino and following our work in #26, we have added some missing fields and changed some defaults.

Besides the `.spec.distribution.modules` key, in the templating phase we need to be able to access also `.metadata.name`, `.spec.distributionVersion`, `.spec.toolsConfiguration` and `.kind`.

Specify (`.spec.region`?) and make available to the templating phase the AWS region in which the cluster is created (which can be different from `.spec.toolsConfiguration.terrafrom.state.s3.region`).

File: furyctl-defaults.yaml (hunk 2 above)
Comment (ralgozino) [reply furyctl-defaults.yaml:53]: internal

File: furyctl-defaults.yaml [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+      certManager:
+        # to create the clusterIssuer, this is an additional clusterIssuer than the two provided by cert-manager, for simplicity
+        clusterIssuer:
+          name: letsencrypt-fury
+          # can be route53 or http01 , if http01 we need to extrapolate the ingress type. If single the class will be nginx, if dual the class will be external
+          type: http01
+          # if auth type is route53, we need to provide the following configurations
+          route53:
```
[the remark this answers is not in the document]
Comment (ralgozino) [reply furyctl-defaults.yaml:61]: internal

File: furyctl-defaults.yaml [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+          ##COOKIE_SECRET is obtained with  `head -c32 /dev/urandom | base64` see https://www.pomerium.io/reference/#cookie-secret
+          COOKIE_SECRET: ""
+          ##IDP_CLIENT_SECRET is the secret configured in the pomerium Dex static client
+          IDP_CLIENT_SECRET: ""
+          ##SHARED_SECRET is obtained with  `head -c32 /dev/urandom | base64` see https://www.pomerium.io/reference/#shared-secret
+          SHARED_SECRET: ""
+      dex:
+        # see dex documentation for more information
+        connectors: []
+    aws:
```
[the remark this answers is not in the document]
Comment (ralgozino) [reply furyctl-defaults.yaml:183]: internal

File: furyctl-defaults.yaml [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+      clusterAutoscaler:
+        nodeGroupAutoDiscovery: asg:tag=k8s.io/cluster-autoscaler/enabled,k8s.io/cluster-autoscaler/example
+        iamRoleArn: arn:aws:iam::123456789012:role/example-cluster-autoscaler
```
[the remark this answers is not in the document]
Comment (ralgozino) [reply furyctl-defaults.yaml:186]: internal

**Pull request merged**